pkg/kubectl/cmd/label.go

```diff
@@ -386,6 +398,9 @@ func labelFunc(obj runtime.Object, overwrite bool, resourceVersion string, label
 	if err != nil {
 		return err
 	}
+	if err = validateSameValue(accessor, labels); err != nil {
```

Comment:
Right, but this should happen *only* when you are not forcing overwrite, which basically means inside the `if !overwrite {` block. Honestly I'd do that check from `validateSameValue` be part of `validateNoOverwrites`, especially that it's already iterating over labels.

Reply:
@soltysh I know what you mean, but it will only make the error info more clearly. What I was trying to fix is the diffrence between `label=nginx --overwrite --dry-run` and `label=nginx --overwrite`(label a same value with --overwrite without --dryrun print 'not labeled'). `validateSameValue` try to solve the problem just by error out, or check this solution [commit/4f924b7](https://github.com/kubernetes/kubernetes/commit/4f924b7d893f521d73a23c5c490943c748148d9a)

Reply:
@soltysh I add a deep equal comparison in the dry-run block as @juanvallejo 's suggestion. PTAL.